src/sema/statements.rs

```diff
@@ -1174,6 +1174,35 @@ fn revert_named_arg(
     }
 }
 
+fn process_event(
```

Comment:
The name of the function should describe what it does or at least deserves a comment.

Reply:
Does the new closure `to_stmt` need a comment?